code/modules/recycling/disposal/loafer.dm

```diff
@@ -0,0 +1,155 @@
+			icon_state = "ploaf3"
+			force = 44
+			throwforce = 44
+			throw_range = 4
+			reagents.add_reagent("radium",25)
+
+		if (7)
+			name = "neutron loaf"
+			desc = "Oh good, the flavor atoms in this prison loaf have collapsed down to a a solid lump of neutrons."
+			icon_state = "ploaf4"
+			force = 66
+			throwforce = 66
+			throw_range = 3
+			reagents.add_reagent("polonium",25)
+
+		if(8)
+			name = "quark loaf"
+			desc = "This nutritional loaf is collapsing into subatomic flavor particles. It is unfathmomably heavy."
+			icon_state = "ploaf5"
+			force = 88
+			throwforce = 88
+			throw_range = 2
+			reagents.add_reagent("bluespace",25)
+
+		if(9)
+			name = "degenerate loaf"
+			desc = "You should probably call a physicist."
+			icon_state = "ploaf6"
+			force = 110
+			throwforce = 110
+			throw_range = 1
+			reagents.add_reagent("bluespace",50)
+
+		if(10)
+			name = "strangelet loaf"
+			desc = "You should probably call a priest."
+			icon_state = "ploaf7"
+			force = 220
+			throwforce = 220
+			throw_range = 0
+			reagents.add_reagent("bluespace",100)
+			START_PROCESSING(SSobj, src) // good job, pro loaf science
+
+/obj/item/reagent_containers/food/snacks/prison_loaf/process()
+	..()
+	if(get_turf(loc))
+		var/edge = get_edge_target_turf(src, pick(alldirs))
+		throw_at(edge, 100, 1)
+
+/obj/item/reagent_containers/food/snacks/prison_loaf/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	. = ..()
+
+/obj/structure/disposalpipe/loafer
+	name = "disciplinary loaf processor"
+	desc = "A pipe segment designed to convert detritus into a nutritionally-complete meal for inmates."
+	icon_state = "pipe-loaf0"
+
+/obj/structure/disposalpipe/loafer/transfer(obj/structure/disposalholder/H)
+	icon_state = "pipe-loaf1"
+	playsound(src.loc, "sound/machines/juicer.ogg", 50, 1)
+	var/obj/item/reagent_containers/food/snacks/prison_loaf/newLoaf = new(H)
+	for(var/atom/movable/A in H)
+		if(A.reagents)
+			A.reagents.trans_to(newLoaf, 1000)
+		if(istype(A, /obj/item/reagent_containers/food/snacks/prison_loaf)
+			var/obj/item/reagent_containers/food/snacks/prison_loaf/otherLoaf = A
+			newLoaf.loaf_factor += otherLoaf.loaf_factor * 1.2
+			newLoaf.loaf_recursion = otherLoaf.loaf_recursion + 1
+		if(istype(A, /mob/living))
+			var/mob/living/poorSoul = A
+			if (issilicon(poorSoul))
+				newLoaf.reagents.add_reagent("oil",10)
+				newLoaf.reagents.add_reagent("silicon",10)
+				newLoaf.reagents.add_reagent("iron",10)
+			else
+				newLoaf.reagents.add_reagent("blood",10)
+				newLoaf.reagents.add_reagent("liquidgibs",10)
+
+			if(ishuman(poorSoul))
+				newLoaf.loaf_factor += (newLoaf.loaf_factor / 5) + 50
+			else
+				newLoaf.loaf_factor += (newLoaf.loaf_factor / 10) + 50
+			poorSoul.emote("scream")
+			poorSoul.die()
+		if(istype(A, /obj/item))
+			var/obj/item/I = A
+			newLoaf.loaf_factor += I.w_class * 5
+		else
+			newLoaf.loaf_factor++
+		qdel(A)
```

Comment:
Feed changeling to prisoners! Or your target!

Reply:
the chef has access to a gibber, so